lib/client/client.go

```diff
@@ -250,18 +250,25 @@ func (proxy *ProxyClient) reissueUserCerts(ctx context.Context, cachePolicy Cert
 		return nil, trace.Wrap(err)
 	}
 
-	key.Cert = certs.SSH
-	key.TLSCert = certs.TLS
-	if params.KubernetesCluster != "" {
-		key.KubeTLSCerts[params.KubernetesCluster] = certs.TLS
-	}
-	if params.RouteToDatabase.ServiceName != "" {
-		key.DBTLSCerts[params.RouteToDatabase.ServiceName] = certs.TLS
-	}
-	if params.RouteToApp.Name != "" {
+	key.ClusterName = params.RouteToCluster
+
+	// Only update the parts of key that match the usage.
+	//
+	// This prevents us from overwriting the top-level key.TLSCert with
+	// usage-restricted certificates.
+	switch params.usage() {
+	case proto.UserCertsRequest_All:
```

Comment:
The `All` indication may be confusing as it could be taken to indicate a possibility of issuing a cert for multiple services if the desired app/DB/k8s-cluster name is set (similar to the original behaviour).

Ideally the proto enum variant would be documented and/or renamed (e.g. to `Basic` or `Generic`? given that it's the fallback in the switch inside `params.usage()`) to avoid this misunderstanding.

Reply:
Good point.
We can't rename the enums without a migration.
I've documented the different `CertUsage` enum values instead.